src/main/src/main/java/org/vfny/geoserver/global/GeoServerFeatureSource.java

```diff
@@ -210,7 +211,7 @@ protected Query makeDefinitionQuery(Query query, SimpleFeatureType schema) throw
             Query defQuery = new Query(query);
             defQuery.setFilter(filter);
             defQuery.setPropertyNames(propNames);
-
+            defQuery.setCoordinateSystem(query.getCoordinateSystem());
```

Comment:
setting the CRS of Query to use Geotools capability of skipping re-projection if CRS is remotely supported